packages/components/app/styles/mixins/_button.scss

```diff
@@ -0,0 +1,288 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+$hds-button-sizes: ( "small", "medium", "large" );
+$hds-button-border-radius: 5px;
+$hds-button-border-width: 1px;
+$hds-button-focus-border-width: 3px;
+
+// these values later may come from the design tokens
+$size-props: (
+  "small": (
+    "font-size": 0.8125rem, // 13px;
+    "line-height": 0.875rem, // 14px - we need to make it even (so we set it slighly larger than the font-size; notice: in Figma is 12px but this would cut some ascendants/descendants)
+    "min-height": 1.75rem,  // 28px
+    "padding": 0.375rem 0.6875rem, // 6px 11px - here we're taking into account the 1px border
+    "padding-dropdown": 0.375rem 0.375rem 0.375rem 0.6875rem, // 6px 6px 6px 11px - here we're taking into account the dropdown chevron
```

Comment:
not super keen on this "variant" approach; I preferred more the local/contextual override (with comment) in the `dropdown.scss` file (but I imagine with this new approach based on the mixin, it may not be possible anymore)